pkgs/applications/editors/vscode/with-extensions.nix

```
@@ -61,15 +65,23 @@ runCommand "${wrappedPkgName}-with-extensions-${wrappedPkgVersion}" {
   dontPatchELF = true;
   dontStrip = true;
   meta = vscode.meta;
-} ''
+} (if system == "x86_64-darwin" then ''
+  mkdir -p $out/bin/
+  mkdir -p "$out/Applications/${longName}.app/Contents/MacOS"
+
+  for path in PkgInfo Frameworks Resources _CodeSignature Info.plist; do
+    cp -r "${vscode}/Applications/${longName}.app/Contents/$path" "$out/Applications/${longName}.app/Contents/"
```

let's just link these paths, otherwise it would end up duplicating the package closure size
```suggestion
    ln -s "${vscode}/Applications/${longName}.app/Contents/$path" "$out/Applications/${longName}.app/Contents/"
```

btw, not blocking for this PR, can we replace this `runCommand` with `buildEnv`?

---

I'm having some trouble testing this with `nix-review wip` -- not sure if I'm not understanding the tool properly, or what. I'm not sure if macOS will handle those directories being a symlink though.

`nix-review wip` run (with the suggested change applied):
<details>
```
❯ nix-shell -p nix-review --run 'nix-review wip'
$ git fetch --force https://github.com/NixOS/nixpkgs master:refs/nix-review/0
$ git worktree add /Users/amanda/.cache/nix-review/rev-75537715afe7df4e599da483399e63aabb358b66-dirty-3/nixpkgs 22b0be560914b738e5342148caebd5c575b5a0b9
Preparing worktree (detached HEAD 22b0be56091)
Updating files: 100% (20075/20075), done.
HEAD is now at 22b0be56091 Merge pull request #70509 from doronbehar/package-snzip
$ nix-env -f /Users/amanda/.cache/nix-review/rev-75537715afe7df4e599da483399e63aabb358b66-dirty-3/nixpkgs -qaP --xml --out-path --show-trace
error: while querying the derivation named 'libcutl-1.10.0':
while evaluating the attribute 'majmin' of the derivation 'libcutl-1.10.0' at /Users/amanda/.cache/nix-review/rev-75537715afe7df4e599da483399e63aabb358b66-dirty-3/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:191:11:
invalid regular expression '([[:digit:]].[[:digit:]]*+).*', at /Users/amanda/.cache/nix-review/rev-75537715afe7df4e599da483399e63aabb358b66-dirty-3/nixpkgs/pkgs/development/libraries/libcutl/default.nix:21:28
Applying `nixpkgs` diff...
error: patch failed: pkgs/applications/editors/vscode/with-extensions.nix:70
error: pkgs/applications/editors/vscode/with-extensions.nix: patch does not apply
Failed to apply diff in /Users/amanda/.cache/nix-review/rev-75537715afe7df4e599da483399e63aabb358b66-dirty-3/nixpkgs
$ git worktree prune
```
</details>

---

it's an error unrelated to this PR. I've created another issue https://github.com/NixOS/nixpkgs/issues/71544.

---

@AmandaCameron are you still hitting the same error on `nix-review wip` ?

---

I tried committing the change instead of just using it from a dirty tree, but now it's just saying "No diffs detected"? I'm confused, and not sure if I'm just overly tired today or if nix-review wip isn't behaving how I'd think it should?

---

>[`nix-review wip` is] just saying "No diffs detected"?

I you staged your changes, you need to add `--staged` flag